packages/server-renderer/src/renderToString.ts

```diff
@@ -83,7 +83,19 @@ function unrollBuffer(buffer: ResolvedSSRBuffer): string {
   return ret
 }
 
-export async function renderToString(input: App | VNode): Promise<string> {
+export interface SSRContext {
+  portals: Record<string, string>
+}
+let ssrContext: SSRContext
```

Comment:
We can't have `ssrContext` here because `renderToString` is async. Concurrent calls may overwrite the context being used. The context must be created inside the call and passed down.

Reply:
okay